vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ModuleManager.java

```diff
@@ -89,4 +99,32 @@ public ISequence checkIfSequenceExists(String moduleId, String sequenceId) {
     public Page<IModule> findByNameOrDescription(Pageable requestedPage,String searchText) {
         return moduleRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage,searchText,searchText);
     }
+    
+    @Override
+    public void deleteModule(String moduleId) {
+        List<ModuleLink> moduleLinksToPersist = new ArrayList<ModuleLink>();
+        if(moduleId != null) {
```

Comment:
turn this logic around and return if true to get rid of this huge if block
```
if (moduleId == null) ...
```

Reply:
Changed